test-utils/src/main/java/com/criteo/publisher/mock/MockedDependenciesRule.java

```diff
@@ -224,9 +226,17 @@ public void resetAllDependencies() {
 
     setUpDependencyProvider();
     setUpCdbMock();
+    setupAndroidUtilMock();
     injectDependencies();
   }
 
+  private void setupAndroidUtilMock() {
+    AndroidUtil androidUtil = new AndroidUtil(dependencyProvider.provideApplication());
```

Comment:
Instead of instantiating the AndroidUtil, why don't you ask the dependency provider for one ?